src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs

```diff
@@ -0,0 +1,2736 @@
+{
+    public int a;
+    public int b;
+
+    public NewStruct(int a, int b)
+    {
+        this.a = a;
+        this.b = b;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (!(obj is NewStruct))
+        {
+            return false;
+        }
+
+        var other = (NewStruct)obj;
+        return a == other.a &&
+               b == other.b;
+    }
+
+    public override int GetHashCode()
+    {
+        var hashCode = 2118541809;
+        hashCode = hashCode * -1521134295 + a.GetHashCode();
+        hashCode = hashCode * -1521134295 + b.GetHashCode();
+        return hashCode;
+    }
+
+    public void Deconstruct(out int a, out int b)
+    {
+        a = this.a;
+        b = this.b;
+    }
+
+    public static implicit operator (int a, int b) (NewStruct value)
+    {
+        return (value.a, value.b);
+    }
+
+    public static implicit operator NewStruct((int a, int b) value)
+    {
+        return new NewStruct(value.a, value.b);
+    }
+}";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertTupleToStruct)]
+        public async Task ConvertFromType3()
+        {
+            var text = @"
+class Test
+{
+    (int a, int b) Method()
+    {
+        [||](int a, int b) t1 = (a: 1, b: 2);
+        (int b, int a) t2 = (b: 1, a: 2);
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    NewStruct Method()
+    {
+        {|Rename:NewStruct|} t1 = new NewStruct(a: 1, b: 2);
+        (int b, int a) t2 = (b: 1, a: 2);
+    }
+}
+
+internal struct NewStruct
+{
+    public int a;
+    public int b;
+
+    public NewStruct(int a, int b)
+    {
+        this.a = a;
+        this.b = b;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (!(obj is NewStruct))
+        {
+            return false;
+        }
+
+        var other = (NewStruct)obj;
+        return a == other.a &&
+               b == other.b;
+    }
+
+    public override int GetHashCode()
+    {
+        var hashCode = 2118541809;
+        hashCode = hashCode * -1521134295 + a.GetHashCode();
+        hashCode = hashCode * -1521134295 + b.GetHashCode();
+        return hashCode;
+    }
+
+    public void Deconstruct(out int a, out int b)
+    {
+        a = this.a;
+        b = this.b;
+    }
+
+    public static implicit operator (int a, int b) (NewStruct value)
+    {
+        return (value.a, value.b);
+    }
+
+    public static implicit operator NewStruct((int a, int b) value)
+    {
+        return new NewStruct(value.a, value.b);
+    }
+}";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertTupleToStruct)]
+        public async Task ConvertFromType4()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        (int a, int b) t1 = (a: 1, b: 2);
+        [||](int a, int b) t2 = (a: 1, b: 2);
+    }
```

Comment:
>} [](start = 4, length = 1)

What if this code included `Write(t1.Item1);`?

Reply:
That code will break.  It's an interesting case... but i consider that acceptable.  if you're converting over to a named struct, then having the 'dual' field naming that tuples provide isn't something we can maintain.

I'm ok with that.

Reply:
Should we detect and report a conflict?

---
In reply to: [201832736](https://github.com/dotnet/roslyn/pull/28257#discussion_r201832736) [](ancestors = 201832736)

Reply:
it would be pretty intensely expensive to do so.  we would have to find all usages and example them.  consider, for example if you had: ```(int a, int b) GetWhatever()```.  And you made that tuple into a named struct.  We'd have to FAR GetWhatever(), then see how the result of it was used.  

I personally don't think it is worth it to do.